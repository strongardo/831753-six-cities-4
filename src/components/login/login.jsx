import React, {createRef} from "react";
import PropTypes from "prop-types";
import {connect} from "react-redux";
import {setUserStatusAsync} from "../../reducer/user/user.js";
import Header from "../header/header.jsx";

const Login = (props) => {
  const loginRef = createRef();
  const passwordRef = createRef();
  const onSubmit = props.onSubmit;

  return (
    <div className="page page--gray page--login">
      <Header />
      <main className="page__main page__main--login">
        <div className="page__login-container container">
          <section className="login">
            <h1 className="login__title">Sign in</h1>
            <form className="login__form form"
              action="#"
              method="post"
              onSubmit={(evt) => {
                evt.preventDefault();
                onSubmit(loginRef.current.value, passwordRef.current.value);
              }}>
              <div className="login__input-wrapper form__input-wrapper">
                <label className="visually-hidden">E-mail</label>
                <input className="login__input form__input" ref={loginRef} type="email" name="email" placeholder="Email" required />
              </div>
              <div className="login__input-wrapper form__input-wrapper">
                <label className="visually-hidden">Password</label>
                <input className="login__input form__input" ref={passwordRef} type="password" name="password" placeholder="Password" required />
              </div>
              <button className="login__submit form__submit button" type="submit">Sign in</button>
            </form>
          </section>
          <section className="locations locations--login locations--current">
            <div className="locations__item">
              <a className="locations__item-link" href="#">
                <span>Amsterdam</span>
              </a>
            </div>
          </section>
        </div>
      </main>
    </div>
  );
};

Login.propTypes = {
  onSubmit: PropTypes.func.isRequired,
};

const mapDispatchToProps = (dispatch) => ({
  onSubmit(login, password) {
    dispatch(setUserStatusAsync({
      login,
      password,
    }));
  },
});

export {Login};
export default connect(null, mapDispatchToProps)(Login);
