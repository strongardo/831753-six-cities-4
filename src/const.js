export const SortType = {
  POPULAR: `Popular`,
  LOW_TO_HIGH: `Price: low to high`,
  HIGH_TO_LOW: `Price: high to low`,
  TOP_RATED_FIRST: `Top rated first`,
};

export const UserStatus = {
  AUTH: `AUTH`,
  NO_AUTH: `NO_AUTH`,
};

export const AppRoute = {
  LOGIN: `/login`,
  OFFER: `/offer/`,
  ELECT: `/elect`,
  ROOT: `/`,
};
