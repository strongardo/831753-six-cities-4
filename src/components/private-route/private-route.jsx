import React from "react";
import PropTypes from "prop-types";
import {Route, Redirect} from "react-router-dom";
import {connect} from "react-redux";
import {AppRoute} from "../../const.js";
import {UserStatus} from "../../const.js";
import {getUserStatus} from "../../reducer/user/selectors.js";


const PrivateRoute = (props) => {
  const {children, userStatus, path, exact} = props;

  if (userStatus !== UserStatus.AUTH && path === AppRoute.ELECT) {
    return <Redirect to={AppRoute.LOGIN} />;
  }
  if (userStatus === UserStatus.AUTH && path === AppRoute.LOGIN) {
    return <Redirect to={AppRoute.ROOT} />;
  }
  return (
    <Route
      path={path}
      exact={exact}
    >
      {children}
    </ Route>
  );
};

PrivateRoute.propTypes = {
  userStatus: PropTypes.string.isRequired,
  exact: PropTypes.bool.isRequired,
  path: PropTypes.string.isRequired,
  children: PropTypes.object.isRequired,
};

const mapStateToProps = (state) => ({
  userStatus: getUserStatus(state),
});


export {PrivateRoute};
export default connect(mapStateToProps)(PrivateRoute);
