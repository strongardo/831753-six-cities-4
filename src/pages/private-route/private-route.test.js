import React from "react";
import renderer from "react-test-renderer";
import {PrivateRoute} from "./private-route.jsx";
import {Login} from "../login/login.jsx";
import {Router} from 'react-router-dom';
import history from '../../history.js';
import {Provider} from "react-redux";
import configureStore from "redux-mock-store";
import {AppRoute, UserStatus} from "../../const.js";
import NameSpace from "../../reducer/name-space.js";

const mockStore = configureStore([]);

it(`Render PrivateRoute`, () => {
  const store = mockStore({
    [NameSpace.USER]: {
      userStatus: UserStatus.AUTH,
      userData: {
        avatarUrl: `/img/avatar-angelina.jpg`,
        email: `mail@ln.kg`,
        isPro: false,
        name: ``,
      },
    },
  });

  const tree = renderer
    .create(
        <Provider store={store}>
          <Router history={history}>
            <PrivateRoute
              userStatus={UserStatus.NO_AUTH}
              exact={true}
              path={AppRoute.ELECT}
            >
              <Login onSubmit={() => {}} />
            </PrivateRoute>
          </Router>
        </Provider>
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
