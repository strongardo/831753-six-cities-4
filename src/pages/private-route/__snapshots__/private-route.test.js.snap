// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render PrivateRoute 1`] = `null`;
